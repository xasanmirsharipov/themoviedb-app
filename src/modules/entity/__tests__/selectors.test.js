import React from 'react';
import Selectors from '../selectors';

describe('Entity Selectors', () => {

	const entity = 'movie';
	const entityType = 'movieEntityType';
	const id = 1;
	const state = {
		entities: {
			[entity]: {
				1: {
					id: 1,
					name: 'Movie 1'
				},
				2: {
					id: 2,
					name: 'Movie 2'
				}
			}
		},
		entity: {
			[entity]: {
				[entityType]: {
					isFetched: true,
					ids: [2],
					meta: {
						totalCount: 0,
						pageCount: 1,
						currentPage: 1,
						perPage: 20
					}
				},
				[`${entityType}One`]: {
					isFetched: true,
					id
				}
			}
		}
	};

	it('getAll selector', () => {

		const props = { entity, name: entityType };
		const selector = Selectors.getAll();

		expect(selector(state, props)).toEqual({
			isFetched: true,
			items: [
				{
					id: 2,
					name: 'Movie 2'
				}
			],
			meta: {
				totalCount: 0,
				pageCount: 1,
				currentPage: 1,
				perPage: 20
			}
		});

		expect(selector(state, props)).toMatchSnapshot();
	});

	it('getOne selector', () => {
		const props = { entity, name: entityType, id };
		const selector = Selectors.getOne();

		expect(selector(state, props)).toEqual({
			isFetched: true,
			item: state.entities[entity][props.id]
		});

		expect(selector(state, props)).toMatchSnapshot();
	});

});