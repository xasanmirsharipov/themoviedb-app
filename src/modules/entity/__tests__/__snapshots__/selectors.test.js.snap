// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity Selectors getAll selector 1`] = `
Object {
  "isFetched": true,
  "items": Array [
    Object {
      "id": 2,
      "name": "Movie 2",
    },
  ],
  "meta": Object {
    "currentPage": 1,
    "pageCount": 1,
    "perPage": 20,
    "totalCount": 0,
  },
}
`;

exports[`Entity Selectors getOne selector 1`] = `
Object {
  "isFetched": true,
  "item": Object {
    "id": 1,
    "name": "Movie 1",
  },
}
`;
